cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -111,17 +111,28 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 
 func removeOldDNSDeploymentIfAnotherDNSIsUsed(cfg *kubeadmapi.ClusterConfiguration, client clientset.Interface, dryRun bool) error {
```

Comment:
hoping that one day we can deprecate kube-dns.
we hardly support it in terms of being in contact with the devs and i haven't seen users since coredns went GA.